src/tests/lxc-test-lxc-attach

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+
+# lxc: linux Container library
+
+# Authors:
+# Christian Brauner <christian.brauner@mailbox.org>
+#
+# This is a test script for the lxc-attach program. It tests whether I/O
+# redirection works correctly.
+
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+set -e
+
+FAIL() {
+	echo -n "Failed " >&2
+	echo "$*" >&2
+        lxc-destroy -n busy -f
+	exit 1
+}
+
+# Create a container, start it and wait for it to be in running state
+lxc-create -t busybox -n busy || FAIL "creating busybox container"
+lxc-start -n busy -d || FAIL "starting busybox container"
+lxc-wait -n busy -s RUNNING || FAIL "waiting for busybox container to run"
+
+# Test if simple attach is working
+attach=$(lxc-attach -n busy -- hostname) # expect output
+if [ "$attach" != "busy" ]; then
+        FAIL " simple attach"
+fi
+
+# Test if we still receive output when stdin does not refer to a tty
+attach=$(lxc-attach -n busy -- hostname < /dev/null) # expect output
+if [ "$attach" != "busy" ]; then
+        FAIL " with stdin redirection < /dev/null"
+fi
+
+# Test if we are silent when stdout does not refer to a tty
+attach=$(lxc-attach -n busy -- hostname > /dev/null) # expect no output
```

Comment:
Adding that would be useful too indeed (as a test for this regression). But I also meant the other way round:

    attach=$(lxc-attach -n busy -- hostname 2>&1)
    if [ -n "$attach" ]; then
        FAIL "unexpected stderr output witih stdout redirection"